src/atc/delta/delta_handle.py

```diff
@@ -99,6 +103,88 @@ def recreate_hive_table(self):
         self.drop()
         self.create_hive_table()
 
-    def merge(self):
-        # This should be a method which can merge from delta to delta
-        raise NotImplementedError
+    def get_tablename(self) -> str:
+        return self._name
+
+    def upsert(
+        self,
+        df: DataFrame,
+        join_cols: List[str],
+        incremental_load: bool = True,
+    ) -> Union[DataFrame, None]:
+        def do_simple_save(
+            df_source: DataFrame,
+            incremental_load_local: bool = True,
+        ):
+            mode = "append" if incremental_load_local else "overwrite"
+
+            self.write_or_append(df_source, mode=mode)
+            print(
+                "Incremental Base - incremental load with append"
+                if incremental_load_local
+                else "Incremental Base - full load with append"
+            )
+            return df_source
+
+        if df is None:
+            return None
+
+        # check null keys in our dataframe.
+        any_null_keys = len(
+            df.filter(" OR ".join(f"({col} is NULL)" for col in join_cols)).take(1)
```

Comment:
If len of this filter is more than 0, the filter is applied, so why not just run the filter on data directly?

Reply:
I have removed this check. And discards rows with null in join cols.